src/requests/consumption.js

```diff
@@ -420,7 +426,638 @@ async function convertRecommendedRolesForProjects(recommendedFor) {
+		const base64 = Buffer.from(response.data, 'binary').toString('base64')
+
+		// Get the content type (e.g., image/jpeg) from the response headers
+		const contentType = response.headers['content-type']
+
+		// Create the Base64 Data URL
+		const base64DataUrl = `data:${contentType};base64,${base64}`
+
+		return base64DataUrl
+	} catch (error) {
+		console.error('Error downloading or converting file:', error.message)
+		throw error
+	}
+}
+
+/**
+ * Publish the Program
+ * @name publishProjectTemplates
+ * @param {Object} programData - Program template data
+ * @returns {Object} - Response of Program creation
+ */
+const publishProgram = function (programData) {
+	return new Promise(async (resolve, reject) => {
+		const result = { success: false, templateId: null, error: null }
+		try {
+			// Format the program template
+			let formattedTemplate = formatProgramTemplate(programData)
+			if (!formattedTemplate.success) {
+				throw new Error('FAILED_TO_FORMAT_TEMPLATE')
+			}
+
+			let template = formattedTemplate.template
+			// fetch the resource details to create
+			const resourceDetailsCreateResponse = await rolloutService.details(template?.resourceDetails?.rolloutId)
+			const resourceDetailsCreate = resourceDetailsCreateResponse.result
+			const programScope = template.scope
+			delete template.resourceDetails
+			let result = {}
+			let programId = template?._id
+
+			// Insert the template into the database
+			const programsCollection = mongoDb.collection(COLLECTIONS.PROGRAMS)
+			// if program is already created , update scope , start and end dates  else create a new program
+			if (programId) {
+				const updateTemplate = {
+					scope: formattedTemplate.template.scope,
+					endDate: formattedTemplate.template.endDate,
+					startDate: formattedTemplate.template.startDate,
+				}
+
+				result = await programsCollection.updateOne(
+					{ _id: template?._id },
+					{
+						$set: updateTemplate,
+					}
+				)
+
+				programId = template?._id
+			} else {
+				result = await programsCollection.insertOne(template)
+				// Validate the result of the template creation
+				if (!result || !result.insertedId) {
+					throw new Error('Failed to insert the template into the database.')
+				}
+				programId = result.insertedId
+			}
+			let solutions = []
+
+			if (resourceDetailsCreate?.status == common.ROLLOUT_STATUS_PENDING && resourceDetailsCreate?.published_id) {
+				const updateTemplate = {
+					scope: formattedTemplate.template.scope,
+					endDate: formattedTemplate.template.endDate,
+					startDate: formattedTemplate.template.startDate,
+				}
+				const solutionsCollection = mongoDb.collection(COLLECTIONS.SOLUTIONS)
+				result = await solutionsCollection.updateOne(
+					{ _id: resourceDetailsCreate?.published_id },
+					{
+						$set: updateTemplate,
+					}
+				)
+
+				solutions.push(resourceDetailsCreate?.published_id)
+			} else {
+				const duplicateResource = await duplicateResources(resourceDetailsCreate)
```

Comment:
@adithyadinesh0412 Can u add comments in between is this duplicateResources fucntion create duplicate library template or solution